source/client.c

```diff
@@ -2723,9 +2781,12 @@ static void s_publish_complete(
     if (task_arg->on_complete) {
         task_arg->on_complete(connection, packet_id, error_code, task_arg->userdata);
     }
-    if (task_arg->timeout_arg) {
-        task_arg->timeout_arg->cancelled = true;
+
+    if (task_arg->timeout_wrapper.timeout_task_arg != NULL) {
```

Comment:
`/* NULL out pointers so that, when timeout task runs, it knows that the request is already complete */`